### ISSUE TITLE
DDLDatabase should allow setting input schema

### ISSUE THREAD

**Issue by anuraaga** [jOOQ/jOOQ#7023]
Currently if `inputSchema` is set when using DDLDatabase, it doesn't find any tables and produces no code. When I remove it, the schema `PUBLIC` is assumed. In addition to this, all table names are uppercased.

This means that for example, a comment on a record class

`Setter for <code>mydb.user.id</code>` becomes `Setter for <code>PUBLIC.USER.ID</code>`

This is especially problematic because `forcedType` declarations using the actual db / table names stop working.

**Comment by lukaseder** [comment]
> This is especially problematic because forcedType declarations using the actual db / table names stop working.

The workaround is to make the expressions case insensitive: `(?i:MyScHeMa.MyTaBlE.mYcOlUmN)`

Would you mind showing an example:

- `CREATE TABLE` statement
- `<inputSchema/>` configuration

Also, do you use the `use` command, or something similar to create tables in a specific MySQL schema / database? That would be an interesting case, probably another thing that can only work when simulating the DDL. (see #7034)

The workaround would be to fully qualify the tables upon their creation...

**Comment by anuraaga** [comment]
The DDL is pretty simple, just something like this

```
CREATE TABLE satellite (
  id BIGINT UNSIGNED AUTO_INCREMENT PRIMARY KEY,
  name VARCHAR(255) NOT NULL,
  international_designator VARCHAR(11) COMMENT 'https://en.wikipedia.org/wiki/International_Designator',
  norad_id VARCHAR(5) UNIQUE COMMENT 'https://en.wikipedia.org/wiki/Satellite_Catalog_Number',
  downlink_frequencies BLOB NOT NULL COMMENT 'Frequencies for receiving data from satellite (Hz) - serialized protobuf Int64List',
  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL ON UPDATE CURRENT_TIMESTAMP,
  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
);
```

And using gradle,
```
jooq {
  myDb(sourceSets.main) {
    generator {
      database {
        recordTimestampFields = 'created_at|updated_at'
        inputSchema = 'mydb'
        forcedTypes {
          forcedType {
            name = 'Int64List'
            expression = 'mydb.satellite.downlink_frequencies'
          }
        }
      }
    }
  }
}
```

We don't have any `use` statement, this is the first DDL run when initializing a fresh database. With the normal MySQL code generator, I don't believe the code generation worked correctly without setting `inputSchema`, even though the database name is in the JDBC URL as well.

Making the type expressions case sensitive isn't a big deal so it's an ok workaround, but the weird `PUBLIC` attached to everything with the DDLDatabase when inputSchema isn't specified would stretch the limits of workaround vs hack ;)

**Comment by lukaseder** [comment]
Thanks for the details. Well, the script itself then doesn't seem to be schema specific, it could be installed on any schema (or database, in MySQL), since your tables aren't prefixed with `mydb`. On the other hand, `DDLDatabase` currently doesn't auto-create such a `mydb` schema in H2, so this approach wouldn't work anyway (I've created and implemented #7039 for this).

How about using the code generator's schema mapping feature? 

```
inputSchema = 'PUBLIC'
outputSchema = 'mydb'
```

Details here:
https://www.jooq.org/doc/latest/manual/code-generation/codegen-advanced/codegen-config-catalog-and-schema-mapping

**Comment by lukaseder** [comment]
About the case sensitivity and default casing, I have created a separate issue for this: #7042. We cannot reasonably guess what the intended case would have been from a given SQL script because all vendors are very different. So we must allow users to explicitly specify the default case (or again, you resort to `` `quoting_names` ``, in case of which jOOQ will get it right)

**Comment by lukaseder** [comment]
`SET SCHEMA` and `SET CATALOG` will be supported in jOOQ 3.11 (#7087). This way, having a `USE` command in the installation script will help indicate the right database to jOOQ:

```sql
USE mydb;
```

**Comment by lukaseder** [comment]
I'm closing this as a duplicate of #7087 and others, as I think that all the issues have now been addressed in jOOQ 3.11. Please feel free to open a new issue if you encounter additional problems.
